src/helper.cc

```diff
@@ -1159,16 +1159,16 @@ helperStatefulHandleRead(const Comm::ConnectionPointer &conn, char *, size_t len
 
     srv->roffset += len;
     srv->rbuf[srv->roffset] = '\0';
-    Helper::Xaction *r = srv->requests.front();
     debugs(84, DBG_DATA, Raw("accumulated", srv->rbuf, srv->roffset));
 
-    if (r == nullptr) {
+    if (srv->requests.empty()) {
         /* someone spoke without being spoken to */
         debugs(84, DBG_IMPORTANT, "ERROR: helperStatefulHandleRead: unexpected read from " <<
```

Comment:
I think we should inform the admin about the consequences of this unexpected read:

```suggestion
        debugs(84, DBG_IMPORTANT, "ERROR: Killing helper process after an unexpected read from " <<
```